src/settings/PatronNotices/TokensList/TokensList.test.js

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import { isString } from 'lodash';
+import {
+  screen,
+  render,
+  within,
+} from '@testing-library/react';
+
+import '../../../../test/jest/__mock__';
+
+import tokensProps from '../tokens';
+import TokensList from './TokensList';
+
+jest.mock('@folio/stripes-template-editor', () => ({
+  TokensSection: jest.fn(({
+    selectedCategory,
+    header,
+    tokens = [],
+    loopConfig = {
+      enabled: true,
+      label: 'label',
+      tag: 'tag',
+    },
+    section,
+    onLoopSelect = jest.fn(),
+    // eslint-disable-next-line no-unused-vars
+    onSectionInit = jest.fn(),
+    onTokenSelect = jest.fn(),
+    ...rest
+  }) => (
+    <div section={section} {...rest}>
+      <div data-testid={`headerFor${section}`}>{header}</div>
+      <ul data-testid="availableTokens">
+        {tokens.map(({ token, allowedFor }) => {
+          const disabled = selectedCategory && !allowedFor.includes(selectedCategory);
+
+          return (
+            <li key={token}>
+              <input
+                data-testid={token}
+                type="checkbox"
+                value={token}
+                label={token}
+                disabled={disabled}
+                onChange={onTokenSelect}
+              />
+            </li>
+          );
+        })}
+        { loopConfig.enabled && (
+          <>
+            <hr />
+            <input
+              data-testid="multipleTokens"
+              type="checkbox"
+              value={loopConfig.tag}
+              label={<strong>{loopConfig.label}</strong>}
+              onChange={onLoopSelect}
+            />
+          </>
+        )}
+      </ul>
+    </div>
+  ))
+}));
+
+const testTokensSection = (sectionTestId) => {
+  describe(`View ${sectionTestId} TokensSection`, () => {
+    it(`should render ${sectionTestId} section`, () => {
+      const sectionTestIdValue = screen.getByTestId(sectionTestId);
+
+      expect(sectionTestIdValue).toBeVisible();
+      expect(sectionTestIdValue).toHaveAttribute('section', sectionTestId);
+      expect(within(screen
+        .getByTestId(`headerFor${sectionTestId}`))
+        .getByText(`ui-circulation.settings.patronNotices.${sectionTestId}TokenHeader`))
+        .toBeVisible();
+    });
+
+    tokensProps[sectionTestId].map((token) => {
```

Comment:
for what we are doing that? i tried start tests, and never see `should render checkbox...` test. how `tokensProps` even connected with `TokensList` component? didn't find it in component.

Reply:
`TokensList` component receives list of tokens as prop, so here also `tokensProps` are tested.
And in callback was missed destructuring of `token`, because of this you haven't met `should render checkbox`.
Fixed that already. Thanks for review.

Reply:
If in the end this approach will be approved, i think it would be better to use `forEach`, because in this case we don't need to return a new modified array.